torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -2,10 +2,12 @@
 #include <fmt/format.h>
 #include <torch/csrc/distributed/autograd/rpc_messages/rpc_with_autograd.h>
 #include <torch/csrc/distributed/autograd/utils.h>
+#include <torch/csrc/distributed/rpc/macros.h>
 #include <torch/csrc/distributed/rpc/profiler/remote_profiler_manager.h>
 #include <torch/csrc/distributed/rpc/rref_context.h>
 #include <torch/csrc/distributed/rpc/rref_impl.h>
 #include <torch/csrc/distributed/rpc/rref_proto.h>
+#include <torch/csrc/distributed/rpc/tensorpipe_utils.h>
```

Comment:
Just a quick comment. Please feel free to ignore if this is already in your plan or if I was wrong. 

`tensorpipe_*` files are in `libtorch_python_distributed_sources` and `rref_*` files are in `libtorch_distributed_sources`. In general `libtorch_python` files can depend on `libtorch` files, but not the other way around. 

So, we might need to extract the ctx implementation from `tensorpipe_utils.h` and put that in, maybe, `rpc/utils.h`. And then both TensorPipe and RRef can use that implementation.

see build_variables.bzl for file categorization: https://github.com/pytorch/pytorch/blob/b39eeb07ed2f60a27b60ad32d172b58d31554282/tools/build_variables.bzl